src/main/java/duke/file/Storage.java

```diff
@@ -0,0 +1,100 @@
+package duke.file;
+
+import duke.exceptions.EmptyField;
+import duke.exceptions.IllegalOperation;
+import duke.list.TaskList;
+import duke.ui.MessageBubble;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+public class Storage {
+    static String STORAGE_PATH = "data/dukeData.txt";
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    /**
+     * Load Duke data from local directory and parse the data to a TaskList object.
+     * If no data is found, a new data file will be created and used.
+     *
+     * @return Stored TaskList
+     */
+    public static TaskList loadFile() {
+        try {
+            File f = new File(STORAGE_PATH); // create a File for the given file path
+            if (!f.exists()) {
+                String directory = STORAGE_PATH.substring(0, STORAGE_PATH.lastIndexOf("/"));
+                Files.createDirectories(Paths.get(directory));
+                if (f.createNewFile()) {
+                    MessageBubble.printMessageBubble("New task list database created!");
+                }
+            }
+            Scanner s = new Scanner(f); // create a Scanner using the File as the source
+            TaskList data = new TaskList();
+            while (s.hasNext()) {
+                String temp = s.nextLine();
+                String[] divided = temp.split(" \\| ");
+                boolean isDone = divided[1].equals("1");
+                String taskDescription = divided[2];
+                switch (divided[0]){
+                case "T":
+                    data.addItem(new Todo(taskDescription, isDone), false);
+                    break;
+                case "D":
+                    data.addItem(new Deadline(taskDescription, isDone, divided[3]), false);
+                    break;
+                case "E":
+                    data.addItem(new Event(taskDescription, isDone, divided[3]), false);
+                    break;
+                default:
+                    throw new EmptyField();
+                }
+            }
+            return data;
+        } catch (IOException | EmptyField | IllegalOperation | IllegalArgumentException e) {
```

Comment:
There should be separate catch statements for each exception, otherwise, consider consolidating these into an extracted method instead of a complicated expression.